frontend/services/sources.go

```diff
@@ -265,36 +267,43 @@ func updateAnnotations(annotations map[string]string, reportedName string) map[s
 }
 
 func GetSourceCRD(ctx context.Context, nsName string, workloadName string, workloadKind WorkloadKind) (*v1alpha1.Source, error) {
-	source, err := kube.DefaultClient.OdigosClient.Sources(nsName).List(ctx, metav1.ListOptions{LabelSelector: labels.SelectorFromSet(labels.Set{
-		consts.OdigosNamespaceAnnotation:    nsName,
-		consts.OdigosWorkloadNameAnnotation: workloadName,
-		consts.OdigosWorkloadKindAnnotation: string(workloadKind),
-	}).String()})
+	list, err := kube.DefaultClient.OdigosClient.Sources(nsName).List(ctx, metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(labels.Set{
+			consts.OdigosNamespaceAnnotation:    nsName,
+			consts.OdigosWorkloadNameAnnotation: workloadName,
+			consts.OdigosWorkloadKindAnnotation: string(workloadKind),
+		}).String(),
+	})
 
 	if err != nil {
 		return nil, err
 	}
-	if len(source.Items) == 0 {
+	if len(list.Items) == 0 {
 		return nil, fmt.Errorf(`source "%s" not found`, workloadName)
 	}
-	if len(source.Items) > 1 {
-		return nil, fmt.Errorf(`expected to get 1 source "%s", got %d`, workloadName, len(source.Items))
+	if len(list.Items) > 1 {
+		return nil, fmt.Errorf(`expected to get 1 source "%s", got %d`, workloadName, len(list.Items))
 	}
 
-	crdName := source.Items[0].Name
+	crdName := list.Items[0].Name
 	crd, err := kube.DefaultClient.OdigosClient.Sources(nsName).Get(ctx, crdName, metav1.GetOptions{})
 
 	return crd, err
 }
 
 func createSourceCRD(ctx context.Context, nsName string, workloadName string, workloadKind WorkloadKind) error {
-	err := CheckWorkloadKind(workloadKind)
+	err := CheckWorkloadKindForSourceCRD(workloadKind)
 	if err != nil {
 		return err
 	}
 
 	source, err := GetSourceCRD(ctx, nsName, workloadName, workloadKind)
-	if source != nil && err == nil {
+	if source != nil {
+		// source already exists, do not create a new one
+		return nil
+	}
+	if err != nil && !strings.Contains(err.Error(), "not found") {
```

Comment:
Unless this is something different than what I'm thinking, you can use

```suggestion
	if err != nil && !apierrors.IsNotFound(err) {
```

from `apierrors "k8s.io/apimachinery/pkg/api/errors"`

Reply:
Would that work for a custom not found message? See [here](https://github.com/odigos-io/odigos/blob/81654d1cadfaaf1a26a4aa5b3ebf230d66069568/frontend/services/sources.go#L282)

```go
if len(list.Items) == 0 {
    return nil, fmt.Errorf(`source "%s" not found`, workloadName)
}
```

Reply:
Ah nope, sorry, thought this check was coming directly from the client